src/syntax/Parser.js

```diff
@@ -1412,6 +1429,23 @@ export class Parser {
                               collection, body);
   }
 
+  // The for-on Statement
+  // for  (  { let | var }  identifier  on  expression  )  statement
```

Comment:
It makes sense. ES6 allows const bindings here.

The comment for parseForOfStatement is just out of date.